app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -123,7 +123,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
         switch (requestCode) {
             case CommCareHomeActivity.UPGRADE_APP:
                 if (resultCode == RESULT_CANCELED) {
-                    Toast.makeText(this, R.string.update_canceled, Toast.LENGTH_LONG).show();
+                    UpdateTask task = UpdateTask.getRunningInstance();
+                    if (task != null) {
+                        Toast.makeText(this, R.string.update_canceled, Toast.LENGTH_LONG).show();
+                        task.cancel(true);
```

Comment:
similar to comment above, I think the right modulo here is acutally

UpdateTask.cancelRunningInstanceIfAny()

which is locked internally and returns a true/false for whether there was an existing running task